crates/bindings-csharp/Runtime/Internal/Module.cs

```diff
@@ -218,12 +218,54 @@ public static void RegisterTable<T>()
         moduleDef.RegisterTable(T.MakeTableDesc(typeRegistrar));
     }
 
-    private static byte[] Consume(this Buffer buffer)
+    private static byte[] Consume(this BytesSource source)
     {
-        var len = FFI._buffer_len(buffer);
-        var result = new byte[len];
-        FFI._buffer_consume(buffer, result, len);
-        return result;
+        if (source == BytesSource.INVALID) {
+            return new byte[0];
+        }
+        else
+        {
+            byte[] buffer = new byte[0x20_000];
+            uint written = 0;
+            while (true)
```

Comment:
Haven't used it myself, but it looks like [ArrayBufferWriter](https://learn.microsoft.com/en-us/dotnet/api/system.buffers.arraybufferwriter-1?view=net-8.0) could help reduce some boilerplate here.

Its `GetSpan(extraSize)` performs automatic resizing and returns a buffer to write into.

Reply:
Leaving this for later.